pkg/querier/plan/plan.go

```diff
@@ -78,6 +80,24 @@ func (t QueryPlan) Equal(other QueryPlan) bool {
 	return bytes.Equal(left, right)
 }
 
+func (t QueryPlan) LineFilterExprs() []syntax.LineFilterExpr {
```

Comment:
This function definitely does not belong to the `QueryPlan`. The plan is only a wrapper around `syntax.Expression` to make it serializeable.

Reply:
related: https://github.com/grafana/loki/pull/12037#discussion_r1499421814